spec/unit/promise_spec.rb

```diff
@@ -338,6 +389,343 @@
+
+        promise.fulfill(:foo)
+
+        expect(promise.instance_variable_get(:@observers)).to be_nil
+      end
+    end
+
+    describe 'when called on a fulfilled Promise' do
+      it 'returns self' do
+        promise = Promise.new
+        promise.fulfill(:foo)
+
+        expect(promise.fulfill(:bar)).to equal(promise)
+      end
+
+      it 'can not change the fulfillment value' do
+        promise = Promise.new
+        promise.fulfill(:foo)
+        promise.fulfill(:bar)
+
+        expect(promise).to be_fulfilled
+        expect(promise.value).to equal(:foo)
+      end
+    end
+
+    describe 'when called on a rejected Promise' do
+      it 'returns self' do
+        promise = Promise.new
+        promise.reject(:foo)
+
+        expect(promise.fulfill(:bar)).to equal(promise)
+      end
+
+      it 'can not change the rejection reason' do
+        promise = Promise.new
+        promise.reject(:foo)
+        promise.fulfill(:bar)
+
+        expect(promise).to be_rejected
+        expect(promise.reason).to equal(:foo)
+      end
+
+      it 'does not set a fulfilment value' do
+        promise = Promise.new
+        promise.reject(:foo)
+        promise.fulfill(:bar)
+
+        expect(promise.value).to be_nil
+      end
+    end
+
+    describe 'when the fulfillment value is a pending Promise' do
+      it 'leaves the promise in a pending state' do
+        other = Promise.new
+
+        promise = Promise.new
+        promise.fulfill(other)
+
+        expect(promise).to be_pending
+      end
+
+      it 'sets the given Promise as the source' do
+        other = Promise.new
+
+        promise = Promise.new
+        promise.fulfill(other)
+
+        expect(promise.source).to equal(other)
+      end
+
+      it 'propagates promise fulfillment' do
+        other = Promise.new
+
+        promise = Promise.new
+        promise.fulfill(other)
+
+        other.fulfill(:foo)
+
+        expect(promise).to be_fulfilled
+        expect(promise.value).to equal(:foo)
+      end
+
+      it 'propagates promise rejection' do
+        other = Promise.new
+
+        promise = Promise.new
+        promise.fulfill(other)
+
+        other.reject(:foo)
+
+        expect(promise).to be_rejected
+        expect(promise.reason).to equal(:foo)
+      end
+    end
+
+    describe 'when the fulfillment value is a fulfilled Promise' do
+      it 'fulfills the Promise with the same value' do
+        other = Promise.new
+        other.fulfill(:foo)
+
+        promise = Promise.new
+        promise.fulfill(other)
+
+        expect(promise).to be_fulfilled
+        expect(promise.value).to equal(:foo)
+      end
+
+      it 'works with Promise subclasses' do
+        other = Class.new(Promise).new
+        other.fulfill(:foo)
+
+        promise = Promise.new
+        promise.fulfill(other)
+
+        expect(promise).to be_fulfilled
+        expect(promise.value).to equal(:foo)
+      end
+    end
+
+    describe 'when the fulfillment value is a fulfilled Promise' do
```

Comment:
Looks like this got copied and pasted from the above `describe` without changing `fulfilled` to `rejected`